### PR TITLE
[Cache] Fallback to positional when keyed results are broken

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 3.3
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

Works around https://github.com/krakjoe/apcu/issues/247 ~~and https://github.com/facebook/hhvm/issues/7867~~

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Ready, now with a notice about MemcachedAdapter + TagAwareAdapter.

File: src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php [hunk 1]
```diff
@@ -19,6 +19,18 @@ class MemcachedAdapter extends AbstractAdapter
...
+     * Constructor.
+     *
+     * Using a MemcachedAdapter with a TagAwareAdapter for storing tags is discouraged.
+     * Using a RedisAdapter is recommended instead. If you cannot do otherwise, be aware that:
+     * - the Memcached::OPT_BINARY_PROTOCOL must be enabled
+     *   (that's the default when using MemcachedAdapter::createConnection());
+     * - tags eviction by Memcached's LRU algorithm will break by-tags invalidation;
+     *   your Memcached memory should be large enough to never trigger LRU.
+     *
+     * Using a MemcachedAdapter as a pure items store is fine of course.
```
Comment (fabpot) [inline src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php]: I would remove `of course`

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php (hunk 1 above)
Comment (nicolas-grekas) [inline src/Symfony/Component/Cache/Adapter/MemcachedAdapter.php]: removed

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**